pkg/security/tests/activity_dumps_test.go

```diff
@@ -636,3 +637,113 @@ func TestActivityDumpsAutoSuppression(t *testing.T) {
 		}
 	})
 }
+
+func TestActivityDumpsAutoSuppressionDriftOnly(t *testing.T) {
+	SkipIfNotAvailable(t)
+
+	// skip test that are about to be run on docker (to avoid trying spawning docker in docker)
+	if testEnvironment == DockerEnvironment {
+		t.Skip("Skip test spawning docker containers on docker")
+	}
+	if _, err := whichNonFatal("docker"); err != nil {
+		t.Skip("Skip test where docker is unavailable")
+	}
+	if !IsDedicatedNodeForAD() {
+		t.Skip("Skip test when not run in dedicated env")
+	}
+
+	var expectedFormats = []string{"profile", "protobuf"}
+	var testActivityDumpTracedEventTypes = []string{"exec", "open", "syscalls", "dns", "bind"}
+
+	outputDir := t.TempDir()
+	os.MkdirAll(outputDir, 0755)
+	defer os.RemoveAll(outputDir)
+
+	rulesDef := []*rules.RuleDefinition{
+		{
+			ID:         "test_autosuppression_exec",
```

Comment:
Should we add a test case where `allow_autosuppression: true` and `workload_drift_only: false` and make sure non drift events from workloads that haven't been learnt yet aren't suppressed ? 🤔